agent/sshd/server.go

```diff
@@ -122,13 +123,43 @@ func (s *Server) sessionHandler(session sshserver.Session) {
 
 		os.Chown(pts.Name(), uid, -1)
 
+		hostport := session.RemoteAddr().String()
+
 		logrus.WithFields(logrus.Fields{
 			"user":       session.User(),
 			"pty":        pts.Name(),
-			"remoteaddr": session.RemoteAddr().String(),
+			"remoteaddr": hostport,
 			"localaddr":  session.LocalAddr().String(),
 		}).Info("Session started")
 
+		var host string
+		if !strings.Contains(hostport, "[") {
+			host = strings.Split(hostport, ":")[0]
+		} else {
+			host = strings.Split(strings.Split(hostport, "[")[1], "]")[0]
+		}
```

Comment:
This string manipulation is confusing, we need to simplify this or add a comment block explaining what it does.

Reply:
The purpose is to extract the IP address from a string that may be IPv4 or IPv6 and contains a port suffix.  I'll add a comment block but can't see a simplification.  Suggestions welcome, of course.

Reply:
I think you could use [LastIndexByte](https://golang.org/pkg/strings/#LastIndexByte) to lookup for last `:` and use the string before that, in [ParseIP](https://golang.org/pkg/net/#ParseIP), which is generic over IPv4 and IPv6.

Reply:
But ParseIP doesn't accept IPv6 enclosed in square brackets, so I'd still have to remove those before calling ParseIP.  Please see [https://play.golang.org/p/S8ONQyXNbu2]( https://play.golang.org/p/S8ONQyXNbu2)

Reply:
In fact, the `net` package has a  specific method for this. See the below code:
```go
func newCode(remoteAddr string) {
	host, port, err := net.SplitHostPort(remoteAddr)
	fmt.Println("Extracted IP: ", host, " port: ", port, " err: ", err)
}
```

Reply:
I like that. Simple to understand and does exactly what's needed, so I'll put that in the code.